xmodule/x_module.py

```diff
@@ -213,40 +213,12 @@ class HTMLSnippet:
     html snippet, along with associated javascript and css
     """
 
-    js = {}
-    js_module_name = None
```

Comment:
What do you think about adding `js_module_name = None` to the `MacoTemplateBlockBase` then, since it's the only class the contains the method that uses the attribute?

Reply:
Added it in https://github.com/openedx/edx-platform/pull/31159/commits/0919fc7dabede63137749959dc84eb6d3c70a86f.

Reply:
@Cup0fCoffee, this would be nice, but let's make sure that there are no classes that override `js_module_name` **after** a class inherits from `MakoTemplateBlockBase` and **before** it inherits from `HTMLSnippet` now. If it would, then changing this would override this attribute, as the value set in `MakoTemplateBlockBase` would take precedence in the MRO.

There are a few classes than inherit from both `MakoTemplateBlockBase` and `HTMLSnippet` - e.g. `SequenceBlock`, `RandomizeBlock`, `ConditionalBlock`. In their definitions, I can only see `XmlMixin` and `XModuleToXBlockMixin` between these two classes, but we should each of them to make sure

Reply:
Oh. I've never had to deal with MRO before, so it didn't even cross my mind to check. Here is the list of all the classes that inherit from both:
```
RandomizeBlock
LibraryContentBlock
ConditionalBlock
LTIBlock
PollBlock
SplitTestBlock
SequenceBlock
```

All of them have only `XmlMixin` and/or `XModuleToXBlockMixin` inbetween `MakoTemplateBlockBase` and `HTMLSnippet`.